components/LocationCard/index.js

```diff
@@ -64,6 +101,26 @@ const ReturnLink = styled(Link)`
 export default function LocationCard({ location }) {
   const router = useRouter();
 
+  const [locations, setLocations] = useLocalStorageState("favoriteLocations", {
+    defaultValue: [],
+  });
+
+  const handleEdit = () => {
+    router.push(`/EditForm/${location.id}`);
+  };
+
+  const handleDelete = () => {
+    if (window.confirm("Sure you want to delete this memory?")) {
+      if (locations && locations.length) {
+        const updatedLocations = locations.filter(
+          (loc) => loc.title !== location.title
+        );
+        setLocations(updatedLocations);
+        router.push("/");
+      }
+    }
+  };
```

Comment:
Similarly, you are already passing down the id to the function, but you're not using it. What is more, you filter based on the `title` instead of the `id`. Let's stick to using the `id`.
```suggestion
  const handleDelete = (id) => {
    if (window.confirm("Sure you want to delete this memory?")) {
      if (locations && locations.length) {
        const updatedLocations = locations.filter(
          (loc) => loc.id !== id
        );
        setLocations(updatedLocations);
        router.push("/");
      }
    }
  };
```